### ISSUE TITLE
`BlurFFT` initialised with `filter=None` gives an unbound local error on `device`

### ISSUE THREAD

**Issue by Andrewwango** [deepinv/deepinv#1253]
Because of the code below, when BlurFFT physics is initialised with no filter, the device variable is unset so gives an unbound local error.

Solution: set device as None as fallback

https://github.com/deepinv/deepinv/blob/c815f3a94c7e90b0bf00e3577d6d3aeff398425d/deepinv/physics/blur.py#L702-L728

**Comment by romainvo** [comment]
Hi @Andrewwango, thanks for reporting this! Do you have this minimal example in mind?

```python
import torch
import deepinv as dinv

physics = dinv.physics.BlurFFT(
    img_size=(64,64),
    device="cuda"
)

x = torch.rand((1,1,64,64), device="cuda")
y = physics(x)
```

We could add a line `device=None` above line 702; this would not solve the problem though, as `self.mask` is initialized with `None` if no filter is provided. Do you expect `BlurFFT` to compute the identity if no filter is provided during init? Otherwise, we could throw a more explicit error message. Both are fine with me
